lib/elixir/lib/code/formatter.ex

```diff
@@ -2169,17 +2169,34 @@ defmodule Code.Formatter do
   end
 
   # A literal list is a keyword or (... -> ...)
-  defp last_arg_to_keyword([_ | _] = arg, _list_to_keyword?) do
+  defp last_arg_to_keyword([_ | _] = arg, _list_to_keyword?, _comments) do
     {keyword?(arg), arg}
   end
 
   # This is a list of tuples, it can be converted to keywords.
-  defp last_arg_to_keyword({:__block__, _, [[_ | _] = arg]} = block, true) do
-    if keyword?(arg), do: {true, arg}, else: {false, block}
+  defp last_arg_to_keyword({:__block__, meta, [[_ | _] = arg]} = block, true, comments) do
+    if keyword?(arg) do
+      block_line = line(meta)
+      {{_, arg_meta, _}, _} = hd(arg)
+      first_line = line(arg_meta)
+
+      comment_before_first_line? =
```

Comment:
We don’t need to traverse all comments to compute this. Perhaps the best option is to drop all comments before block_line and then see if the first comment is before the first line. Thoughts?

Reply:
Hi, @josevalim Thank you for your review!
I thought `Enum.any?` would stop immediately when it find the first comment after block line.. .:think: ...

Reply:
> It stops when it finds but it continues when it doesn’t. ;)
I think i know what you said. :)

Reply:
Fixed with 1fc62ce